### PR TITLE
Add option for alternate mod to change how many times the same key can be repeated

### PR DESCRIPTION
**Pull request by ryanbester**
Adds an option in the mod customisation panel to adjust the maximum number of times the same key can be repeated.

By default, this is set to 1 to match the previous behaviour, but can be adjusted by the player.

### CODE REVIEW

**Comment by bdach** [comment]
I'm not sure this sort of setting is something we want or something that users are actually going to use. I for one do not recall something like this being proposed even once until now.

Also there is no test coverage for this (see `TestSceneOsuModAlternate`).

File: osu.Game.Rulesets.Osu/Mods/OsuModAlternate.cs [hunk 1]
```diff
@@ -89,9 +102,22 @@ private bool checkCorrectAction(OsuAction action)
             {
                 // User alternated correctly.
                 lastActionPressed = action;
+                numActions = 0;
                 return true;
             }
 
+            if (lastActionPressed == action)
+            {
+                // If same action, increment counter.
```
Comment (bdach) [inline osu.Game.Rulesets.Osu/Mods/OsuModAlternate.cs:111]: This comment doesn't bring any value, it just restates the code. Comments should not be talking about what is being done, they should be talking about *why* it is being done.

File: osu.Game.Rulesets.Osu/Mods/OsuModAlternate.cs [hunk 2]
```diff
@@ -89,9 +102,22 @@ private bool checkCorrectAction(OsuAction action)
...
+                numActions++;
```
Comment (bdach) [inline osu.Game.Rulesets.Osu/Mods/OsuModAlternate.cs:112]: nit: would probably rename to something like `lastActionPressCount` or something

**Review by bdach — COMMENTED** [review]
I'm personally against merging this as said above, but here's a review of the code that is here right now

**Comment by peppy** [comment]
I honestly don't think this makes sense.

If anything, I'd imagine a setting which only requires alternating after a certain BPM is reached (ie. note density) to allow for using the same key in sparse sections.

Going to close this for now. Feel free to open a discussion to see if the idea gains traction, then we can revisit adding it as a setting.

**Pull request closed without merging**